cwl/src/main/scala/wdl4s/cwl/ExpressionEvaluator.scala

```diff
@@ -0,0 +1,35 @@
+package wdl4s.cwl
+
+import wdl4s.wdl.util.JsUtil
+import wdl4s.wdl.values.WdlValue
+
+// http://www.commonwl.org/v1.0/CommandLineTool.html#Expressions
+object ExpressionEvaluator {
+  // A code fragment wrapped in the $(...) syntax must be evaluated as a ECMAScript expression.
+  private val EcmaScriptRegex = """\$\((.*)\)""".r
```

Comment:
I think if there is significant differences in these (and it seems that there are) we should do

```
type ECMAScriptExpression = ECMAScript :+: ECMAFunction :+: CNil 
type ECMAScript= String Refined MatchesRegex[W.`"""\$\((.*)\)"""`.T]
type ECMAFunction = String Refined MatchesRegex[W.`"""\$\{(.*)\}"""`.T]
```
and let circe do this regex matching for us

Reply:
We could let Circe do more specific matching for us, but it seems we'd still need these near-duplicate regexes to be able to do the captures.

Reply:
Having said that, the existing `MatchesRegex` doesn't match the function.

Reply:
Good point re: captures.  I wonder if Refined has anything for us in that department.

Reply:
I looked and didn't see anything, so right now I'm just trying to fix the existing regex to match functions.